### PR TITLE
chore(root): better cursor background agents fixes NV-7660

### PR DESCRIPTION
**Pull request by merrcury**
## What changed

Cursor background agents were reworked to use a minimal Cursor base image and externalize runtime services and boot logic. The Dockerfile now provides a lightweight agent image with Node 22, pnpm, TypeScript and Docker tooling while startup/install responsibilities were moved to idempotent scripts and a Cursor environment manifest; ClickHouse agent settings were added. The change removes baked-in Mongo/Redis/healthchecks so services are started and orchestrated via docker-compose at runtime.

## Affected areas

- **.cursor (Dockerfile & environment.json)**: Replaced the previous dev image with a Cursor background-agent base image (Node 22, pnpm, TypeScript, fuse-overlayfs, containerd/Docker tooling) and added an environment.json to declare build/start profiles and terminal tasks; removed ENTRYPOINT/HEALTHCHECK/EXPOSE so orchestration is external.
- **.cursor/scripts**: Added install.sh (idempotent dependency install, optional enterprise symlinking, .env seeding, pnpm build) and start.sh (ensure Docker, bring up docker-compose.agent, run ClickHouse local migrations, seed agent user/org).
- **apps/api & apps/ws**: Added ClickHouse connection variables to their .env.agent files to enable analytics/local ClickHouse usage.
- **docker/local**: Adjusted MongoDB volume mapping to use host temp dir (${TMPDIR:-/tmp/mongo} → /data/db).
- **scripts**: Simplified scripts/setup-agent.sh into a thin wrapper that delegates to the new Cursor scripts.
- **Enterprise code**: Novu EE packages are referenced/handled by the new install flow (enterprise symlink logic present).

## Key technical decisions

- Agent image no longer bundles datastore services; MongoDB/Redis/ClickHouse run via docker-compose at runtime to allow flexible orchestration and smaller images.
- Removed Dockerfile ENTRYPOINT/HEALTHCHECK/EXPOSE to avoid baked-in service lifecycle expectations.
- Provisioned container-side Docker tooling and fuse-overlayfs to support Docker-in-Docker use-cases for background agents.
- Install/start split into idempotent scripts to support multi-repo enterprise symlinks and safer reboots.

## Testing

No automated tests were added; manual verification is planned — the author provided a recorded agent run artifact and will spin additional agents to validate edge cases and gaps.

[Review Change Stack](https://app.coderabbit.ai/change-stack/novuhq/novu/pull/11132?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

Novu EE packages are also there. We still need some more testing. I'll spin some more agents to see if there is any gap. 

### Screenshots

Recorded by Cursor from New Agent - https://cursor.com/agents/bc-bdb844c2-e02a-48ca-a757-9ba2cc1a525c/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fnovu_login_demo.mp4

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

**Comment by scopsy** [comment]
@merrcury did you had a chance to test that computer use is working with this base image? (cursor is able to view the computer and use it to record videos, screenshots?)

**Review by scopsy — APPROVED** [review]

**Comment by merrcury** [comment]
@scopsy This is the video made in new setup https://cursor.com/agents/bc-bdb844c2-e02a-48ca-a757-9ba2cc1a525c/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fnovu_login_demo.mp4

I am thinking of trying some more edge cases and see if everything works. It did show me bunch of screenshots as well.

**Comment by scopsy** [comment]
Amazing! Not this scope, but maybe next, we can do a playbook for packacge-enerprise changes with 2 prs, etc.

**Comment by scopsy** [comment]
@cursoragent resolve conflicts

File: apps/ws/src/.env.agent [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+WS_CONTEXT_PATH=
+
+NEW_RELIC_ENABLED=false
+NEW_RELIC_APP_NAME="[LOCAL] - ws"
+NEW_RELIC_APPLICATION_LOGGING_FORWARDING_ENABLED=false
+LOG_LEVEL=info
+
+SOCKET_IO_ADMIN_USERNAME=admin
+# "changeit" encrypted with bcrypt
+SOCKET_IO_ADMIN_PASSWORD_HASH=$2b$10$heqvAkYMez.Va6Et2uXInOnkCT6/uQj1brkrbyG3LpopDklcq7ZOS
```
[the remark this answers is not in the document]
Comment (scopsy) [reply apps/ws/src/.env.agent:24]: This is an agent ephemeral environment, used on testing data

**Comment by scopsy** [comment]
@merrcury should we merge?

**Comment by merrcury** [comment]
> @merrcury should we merge?
 
We should. I tested multiple time. It works. If there is some flakiness I'll do another PR
